src/shims/unix/foreign_items.rs

```diff
@@ -677,9 +677,17 @@ pub trait EvalContextExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
                 let cpusetsize = this.read_target_usize(cpusetsize)?;
                 let mask = this.read_pointer(mask)?;
 
-                // TODO: when https://github.com/rust-lang/miri/issues/3730 is fixed this should use its notion of tid/pid
-                let thread_id = match pid {
-                    0 => this.active_thread(),
+                let thread_id = match (pid, &*this.tcx.sess.target.os) {
+                    (0, _) => this.active_thread(),
+                    (tid, "linux") => {
+                        if let Ok(tid) = this.machine.threads.thread_id_try_from(tid) {
```

Comment:
This will not return the right ID. See the gettid implementation to see how ThreadId is converted to the user-visible TID; this here will have to do the opposite transformation. Please put both directions in some shared place to make sure they remain consistent in the future.